src/App.tsx

```diff
@@ -33,6 +58,60 @@ const App: React.FC = () => {
   const getResult: GetResult = (res) => setResult(res);
   const setIsBlockingToggle = (flag: boolean) => setIsBlocking(flag);
 
+  window.addEventListener('offline', setIsOffline, false);
+  window.addEventListener('online', setIsOnline, false);
+
+  useEffect(() => {
+    setAppLanguage('en');
+  });
+
+  useEffect(() => {
+    if (isOffline) setTimeout(() => offlineContentVisibilityToggle(true), offlineComponentTimeout);
+  }, [isOffline]);
+
+  const askUserToUpdate = (reg: ServiceWorkerRegistration) => {
+    console.log('ddddd');
+    setModaleViewToggle(true);
+    const okButton = document.getElementById('okButton') as HTMLButtonElement;
+    okButton.onclick = () => {
+      reg.waiting?.postMessage({ type: 'SKIP_WAITING' });
+      setModaleViewToggle(false);
+    };
+  };
+
+  window.addEventListener('load', async () => {
```

Comment:
Will we have one more lester on every function call? Possibly this is a memory leak and perforrmance issue